src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -909,6 +923,7 @@ - (void)asyncDispatchToMatterQueue:(dispatch_block_t)block errorHandler:(nullabl
 
 - (void)syncRunOnWorkQueue:(SyncWorkQueueBlock)block error:(NSError * __autoreleasing *)error
 {
+    VerifyOrDie(!chip::DeviceLayer::PlatformMgrImpl().IsWorkQueueCurrentQueue());
```

Comment:
```suggestion
    assertChipStackLockedByCurrentThread();
```
please.

Reply:
I actually want the contrary. I want to make sure we are *not* on the Matter queue to avoid sync waiting on ourself.